### ISSUE TITLE
NewType's type incorrectly reported as type

### ISSUE THREAD

**Issue by adampauls** [microsoft/pyright#8430]
I believe I'm understanding [typeshed](https://github.com/python/typeshed/blob/f863db6bc5242348ceaa6a3bca4e59aa9e62faaa/stdlib/typing.pyi#L316) correctly, `NewType` is its own type (for as of Python 3.11), but Pyright reports it as `type`.

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

```python
from typing import NewType

X = NewType("X", int)
# reveal_type(X)  # Type of "X" is "type[X]"
print(type(X)) # prints <class 'typing.NewType'>
print(isinstance(X, type)) # False
print(isinstance(int, type)) # True
```

**VS Code extension or command-line**
Command line 1.1.371.

**Comment by erictraut** [comment]
I think pyright is working correctly here.

`NewType` effectively creates a new class object with the name `X`. According to the typing spec, it should be treated by type checkers as a subclass of class that is passed to `NewType` (in your example, `int`). Its type is described as `type[X]`. It can be assigned to a variable of type `type[X]` or `type[int]`.

```python
X = NewType("X", int)
reveal_type(X) # type[X]
v1: type[X] = X
v2: type[int] = X
```

Other Python type checkers including mypy agree with this behavior.

For more details, refer to [this section](https://typing.readthedocs.io/en/latest/spec/aliases.html#newtype) of the typing spec.

As with many symbols exported from the `typing` module, there is some magic performed behind the scenes at runtime to implement these constructs, and if you start to poke around with the runtime objects, the facade will crumble. Your use of `isinstance` is an example of this.

**Comment by adampauls** [comment]
Well, it's hard for me to read that spec as saying anything about the type of a NewType itself, other than 
> For a static type checker Derived = NewType('Derived', Base) is roughly equivalent to a definition:
> ```class Derived(Base):
>    def __init__(self, _x: Base) -> None:
> ```

It's certainly very rough! For example, `print(issubclass(X, int))` crashes with `issubclass() arg 1 must be a class`. It seems to me that pyright could respect the spec while making the static behavior of `X` itself match the runtime behavior, but I can see that it might not be worth the effort.
